source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -797,10 +797,10 @@ namespace aspect
       // We receive radius, longitude, latitude and we need to turn it first back into
       // longitude, latitude, depth for internal use, and push_forward to cartesian coordiantes.
       Point<3> position_point;
-      position_point(0) = position_tensor[2];
-      position_point(1) = position_tensor[1];
+      position_point(0) = position_tensor[1];
+      position_point(1) = position_tensor[2];
 
-      const double radius = semi_major_axis_a / (std::sqrt(1 - eccentricity * eccentricity * std::sin(position_point(1)) * std::sin(position_point(1))));
+      const double radius = semi_major_axis_a / (std::sqrt(1 - eccentricity * eccentricity * std::sin(position_point(0)) * std::sin(position_point(0))));
```

Comment:
I do not think this change is correct, since you only need to switch the long and lat intake. I also tested it, and if you remove just this, the new point will be the same as before. 

Looking at it, it might be worth implementing a few more iterations for the pull back function, but that is a separate issue.

Reply:
Agreed! This matches the functions in the reference and returns the same point as before.

Reply:
@MFraters - the documentation was added by me. I have no idea where the original implementation came from, so I had to decide between keeping the description from the literature, modifying the description to fit the code, or rewriting the code. I did check that the two algorithms were equivalent, but I agree that this is not perfect. 

I guess finding the original literature or rewriting the code would be the optimal solutions.

Reply:
The original implementation came from @dsarahstamps (see issue #1950). 

@bobmyhill Thanks for adding that documentation! That made checking this pull request a lot easier :)

I made an issue for the rewrite: #3277 

@bobmyhill, I now also saw your issue #3001. Are you saying there that iterating would not help because it is already the analytical solution?